core-curriculum/knowledge/akka.json

```diff
@@ -489,5 +489,18 @@
         { "question": "How to generate a stream from a graph?" }
       ]
     }
+  ],
+
+  "projects": [
+    {
+      "id": "akka-webcrawler",
+      "name": "Web Crawler using Akka",
+      "description": "The goal of this project is to implement a web crawler using Akka Actors.\n\n# Requirements\n\nDesign and implement an Akka application that crawls through the web parsing html pages. The application should start with a single URL entry point, then parse that HTML page, finding anchor tags to repeat the process.\n\nThings to consider:\n\n* How would you struture the actor hierarchy?\n\n* What should happen when something goes wrong? (Unauthorized Resource, lost connection, etc.)\n\n* The process should end eventually, mind search depth.\n\n* The application should be testable\n\n* As bonus point you could implement this as a cluster.\n\n**Note:** We don't actually care about what's in the pages that you parse, we just care the overall process. You can just discard the data, but feel free to add any type of post processing you want (HTML validation, top 10 most frequent words, accessiblity compliance, etc.)\n\n# Deliverables\n\nA GitHub repository with a working application."
+    },
```

Comment:
This is a fun project :)

I wonder if we should give people some hints on which library for parsing HTML they could use, since that's not really Akka related? And I think it's best to DO specify exactly which task they must execute with the page content.

Reply:
okay, let me fix that.

Reply:
Okay, fixed :)